docs: Update Caching intro (#1764)
cube-js/cube.js: docs/content/Caching/Caching.md

The second level of caching is called **pre-aggregations**, and requires explicit configuration to activate.

We do not recommend changing the default **in-memory** caching configuration

> unless it is necessary. To speed up queries performance, consider using

Reviewer: ```suggestion
unless it is necessary. To speed up query performance, consider using
```

});
```

## In-memory Cache

Cube.js caches the results of executed queries using in-memory cache. The cache
key is a generated SQL statement with any existing query-dependent
pre-aggregations.

Upon receiving an incoming request, Cube.js first checks the cache using this key.
If nothing is found in the cache, the query is executed in the database and the result set
is returned as well as updating the cache.

If an existing value is present in the cache and the `refreshKey` value for
the query hasn't changed, the cached value will be returned. Otherwise, a
[query renewal](#in-memory-cache-force-query-renewal) will be performed.

### Refresh Keys

> Cube.js takes great care to prevent unnecessary queries from hitting your database. The first stage caching system caches query results in Redis (or in the in-memory store in development), but Cube.js needs a way to know if the data powering that query result has changed. If the underlying data isn't any different, the cached result is valid and can be returned skipping that expensive query, but if there is a difference, the query needs to be re-run and its' result cached.

Reviewer: ```suggestion
Cube.js takes great care to prevent unnecessary queries from hitting your database. The first stage caching system caches query results in Redis (or in the in-memory store in development), but Cube.js needs a way to know if the data powering that query result has changed. If the underlying data isn't any different, the cached result is valid and can be returned skipping an expensive query, but if there is a difference, the query needs to be re-run and its result cached.
```

  // By default Cube.js will check this refreshKey every 10 seconds
  refreshKey: {
    sql: `SELECT MAX(created_at) FROM orders`
  }
});
```

By default, Cube.js will check and invalidate the cache in the background when in [development mode][link-development-mode]. When development mode is disabled you can set `CUBEJS_SCHEDULED_REFRESH_TIMER=true` to enable this behavior.

We recommend enabling background cache invalidation in a separate Cube.js worker for production deployments. Please consult the [Production Checklist][link-production-checklist] for more information.

[link-production-checklist]: /deployment/production-checklist
[link-development-mode]: /configuration/overview#development-mode
[link-production-checklist-refresh]: /deployment/production-checklist#set-up-refresh-worker

If background refresh is disabled, Cube.js will refresh the cache during query
execution. Since this could lead to delays in responding to end-users, we
recommend always enabling background refresh.

### Default Refresh Keys

The default values for `refreshKey` are
 * `every: '2 minute'` for BigQuery, Athena, Snowflake, and Presto.
 * `every: '10 second'` for all other databases.

> +You can use a custom SQL query for checking if a refresh is required by changing the [`refreshKey`](/cube#parameters-refresh-key) property in a cube's Data Schema. Often, a `MAX(updated_at_timestamp)` for OLTP data is a viable option, or examining a metadata table for whatever system is managing the data to see when it last ran.

Reviewer: ```suggestion
+You can use a custom SQL query to check if a refresh is required by changing the [`refreshKey`](/cube#parameters-refresh-key) property in a cube's Data Schema. Often, a `MAX(updated_at_timestamp)` for OLTP data is a viable option, or examining a metadata table for whatever system is managing the data to see when it last ran.
```

> ### How to disable the cache?

Reviewer: How about rephrasing this to "Can you disable the cache in Cube.js?"?

There's no straightforward way to disable caching in Cube.js.

> The reason for it is Cube.js is not just stores cached values but uses the cache as a point of synchronization and coordination between nodes in a cluster.

Reviewer: ```suggestion
The reason is that Cube.js not just stores cached values but uses the cache as a point of synchronization and coordination between nodes in a cluster.
```

For the sake of design simplicity, Cube.js doesn't distinguish client invocations, and all calls to the data load API are idempotent.
This provides excellent reliability and scalability but has some drawbacks.

> One of those load data calls can't be traced to specific clients, and as a consequence, there's no guaranteed way for a client to initiate a new data loading query or know if the current invocation wasn't initiated earlier by another client.

Reviewer: Not sure if we need a better transition to this sentence. What is the word "those" referring to? Would it be better to say "Individual load data calls can't be...."?

Author: The whole section needs a rewrite here. In this PR I'm just moving it. I'd recommend leave it as it for now and then rewrite the whole piece in the future PRs.
